src/components/map/OpaqueLayersService.js

```diff
@@ -19,39 +19,37 @@ goog.require('ga_map_service');
       var is3DActive = false;
 
       var unreg = function() {
-        angular.forEach(unregKeys, function(key) {
-          if (key) {
-            ol.Observable.unByKey(key);
-          }
-        });
-        unregKeys = [];
+        var key;
+        while (key = unregKeys.pop()) {
+          ol.Observable.unByKey(key);
+        }
       };
-
       var updateHiddenState = gaDebounce.debounce(function() {
         var sortedLayers = layers.slice().reverse();
         var hide = false;
 
         unreg();
 
-        angular.forEach(sortedLayers, function(layer) {
+        sortedLayers.forEach(function(layer) {
```

Comment:
Some array functions on IE9 are not fully supported. I hope that's not the case for forEach...